terraform/example/Makefile

```diff
@@ -0,0 +1,43 @@
+AWS_ACCESS_KEY_ID  ?= XXXXX
+AWS_SECRET_ACCESS_KEY ?= YYYY
+AWS_DEFAULT_REGION ?= us-east-1
+ENVIRONMENT ?= staging
+
+current_dir := $(shell pwd)
+container_dir := /opt/app
+
+archive:
+	make -C ../../ archive
+init:
+	docker run --rm -ti \
+		-v $(current_dir):$(container_dir) \
+		-e ENVIRONMENT=$(ENVIRONMENT) \
+		-e AWS_ACCESS_KEY_ID=$(AWS_ACCESS_KEY_ID) \
+		-e AWS_SECRET_ACCESS_KEY=$(AWS_SECRET_ACCESS_KEY)\
+		-e AWS_DEFAULT_REGION=$(AWS_DEFAULT_REGION) \
+		--workdir=$(container_dir)\
+		hashicorp/terraform:0.11.1 \
```

Comment:
Can we pull `hashicorp/terraform:latest` instead, and not pin the version in the Terraform provider instantiation? That's less overhead in future.

Reply:
we should fix to a latest known stable version. Make it property driven so that it can be upgraded easily in future.